src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotHaveIncorrectSuffix.cs

```diff
@@ -56,13 +61,182 @@ public abstract class IdentifiersShouldNotHaveIncorrectSuffixAnalyzer : Diagnost
+                    // suffixes, because alone among the suffixes, "Ex" is the only one that
+                    // isn't itself a known type or a language keyword.
+                    if (name.HasSuffix(ExSuffix) && name.Length > ExSuffix.Length)
+                    {
+                        context.ReportDiagnostic(
+                            namedTypeSymbol.CreateDiagnostic(TypeNewerVersionRule, ExSuffix, name));
+                        return;
+                    }
+
+                    if (name.HasSuffix(NewSuffix))
+                    {
+                        context.ReportDiagnostic(
+                            namedTypeSymbol.CreateDiagnostic(TypeNewerVersionRule, NewSuffix, name));
+                        return;
+                    }
+                }, SymbolKind.NamedType);
+
+            analysisContext.RegisterSymbolAction(
+                (SymbolAnalysisContext context) =>
+                {
+                    var methodSymbol = (IMethodSymbol)context.Symbol;
+                    if (methodSymbol.GetResultantVisibility() != SymbolVisibility.Public)
+                    {
+                        return;
+                    }
+
+                    if (methodSymbol.IsOverride)
+                    {
+                        return;
+                    }
+
+                    string name = methodSymbol.Name;
+
+                    if (name.HasSuffix(ExSuffix))
+                    {
+                        context.ReportDiagnostic(
+                            methodSymbol.CreateDiagnostic(MemberNewerVersionRule, ExSuffix, name));
+                        return;
+                    }
+
+                    // We only fire on member suffix "New" if the type already defines
+                    // another member minus the suffix, e.g., we only fire on "MemberNew" if
+                    // "Member" already exists. For some reason FxCop did not apply the
+                    // same logic to the "Ex" suffix, and we follow FxCop's implementation.
+                    if (name.HasSuffix(NewSuffix))
+                    {
+                        string nameWithoutSuffix = name.WithoutSuffix(NewSuffix);
+                        INamedTypeSymbol containingType = methodSymbol.ContainingType;
+
+                        if (MethodNameExistsInHierarchy(nameWithoutSuffix, containingType))
+                        {
+                            context.ReportDiagnostic(
+                                methodSymbol.CreateDiagnostic(MemberNewerVersionRule, NewSuffix, name));
+                            return;
+                        }
+                    }
+
+                    if (name.HasSuffix(ImplSuffix))
+                    {
+                        context.ReportDiagnostic(
+                            methodSymbol.CreateDiagnostic(MemberWithAlternateRule, ImplSuffix, name, CoreSuffix));
+                    }
+                }, SymbolKind.Method);
```

Comment:
Wow, I totally missed that. The FxCop method is `public override ProblemCollection Check(Member member)`, and its implementation does not restrict itself to methods. Good catch. Working on this -- which is your last comment -- now.